src/main/resources/com/google/api/codegen/csharp/gapic_client.snip

```diff
@@ -0,0 +1,465 @@
+        /// and <see cref="CreateAsync(ServiceEndpoint, {@settings.name})"/> will create new channels, which could
+        /// in turn be shut down by another call to this method.</remarks>
+        /// <returns>A task representing the asynchronous shutdown operation.</returns>
+        public static Task ShutdownDefaultChannelsAsync() => s_channelPool.ShutdownChannelsAsync();
+
+        /// <summary>
+        /// The underlying gRPC {@api.grpcServiceName} client.
+        /// </summary>
+        public virtual {@api.grpcTypeName} GrpcClient
+        {
+            get { throw new NotImplementedException(); }
+        }
+
+        @join method : api.apiMethods
+            {@xmlDoc(method.doc)}
+            {@method(method, FALSE)}
+
+        @end
+    }
+@end
+
+@private implClient(api, settings)
+    public sealed partial class {@api.implName} : {@api.name}
+    {
+        private readonly ClientHelper _clientHelper;
+        @join call : api.apiCallableMembers
+            private readonly ApiCall<{@call.requestTypeName}, {@call.responseTypeName}> {@call.name};
+        @end
+
+        public {@api.implName}({@api.grpcTypeName} grpcClient, {@settings.name} settings)
+        {
+            this.GrpcClient = grpcClient;
+            {@settings.name} effectiveSettings = settings ?? {@settings.name}.GetDefault();
+            _clientHelper = new ClientHelper(effectiveSettings);
+            @join call : api.apiCallableMembers
+                {@call.name} = _clientHelper.BuildApiCall<{@call.requestTypeName}, {@call.responseTypeName}>(
+                    GrpcClient.{@call.methodName}Async, GrpcClient.{@call.methodName}, effectiveSettings.{@call.methodName}Settings);
+            @end
+        }
+
+        /// <summary>
+        /// The underlying gRPC {@api.grpcServiceName} client.
+        /// </summary>
+        public override {@api.grpcTypeName} GrpcClient { get; }
+        
+        @join method : api.apiMethodsImpl
+            {@xmlDoc(method.doc)}
+            {@method(method, TRUE)}
+
+        @end
+    }
+@end
+
+@private method(method, isImpl)
+    @switch method.type.toString
+    @case "FlattenedMethod"
+        {@flattenedMethodSync(method, isImpl)}
+    @case "FlattenedAsyncCallSettingsMethod"
+        {@flattenedMethodAsyncCallSettings(method, isImpl)}
+    @case "FlattenedAsyncCancellationTokenMethod"
+        {@flattenedMethodAsyncCancellationToken(method, isImpl)}
+    @case "PagedFlattenedMethod"
+        {@pagedFlattenedMethodSync(method, isImpl)}
+    @case "PagedFlattenedAsyncMethod" 
+        {@pagedFlattenedMethodAsync(method, isImpl)}
+    @default
+        *** ERROR: Cannot handle method type '{@method.type.toString}' ***
+    @end
+@end
+
+@private flattenedMethodSync(method, isImpl)
+    @if isImpl
+        public override {@method.responseTypeName} {@method.name}(
+        {@""}    {@objectParams(method.methodParams)}) => {@method.callableName}.Sync(
```

Comment:
Looks like there are plenty more instances of `{@""}` that can be removed from this file.